File: engine/transpile/src/test/resources/golden/SYK007/WS_ファイル状態.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK007 のデータ部レコード WS-ファイル状態 を逐語対訳した自動生成コード。 */
public final class WS_ファイル状態 {

    /** 01 WS-ファイル状態 総バイト長 2。 */
    public static final int LENGTH = 2;

    private final byte[] data;

    public WS_ファイル状態() {
        this.data = new byte[2];
    }

    public WS_ファイル状態(byte[] source) {
        this.data = Arrays.copyOf(source, 2);
    }

    public byte[] data() {
        return data;
    }

    // WS-STKEXTR-STATUS
    public String get_WS_STKEXTR_STATUS() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 2);
    }
    public void set_WS_STKEXTR_STATUS(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 2, value);
    }
}

File: src/engine/transpile/src/test/resources/golden/SYK007/WS_ファイル状態.py
"""SYK007 のデータ部レコード WS-ファイル状態 を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class WS_ファイル状態:
    """01 WS-ファイル状態 総バイト長 2。"""

    LENGTH = 2

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(2)
        else:
            self._data = bytearray(data)
            if len(self._data) < 2:
                self._data.extend(b"\x00" * (2 - len(self._data)))

    @property
    def data(self):
        return self._data

    # WS-STKEXTR-STATUS
    def get_WS_STKEXTR_STATUS(self):
        return decode_alphanumeric(self._data, 0, 2)
    def set_WS_STKEXTR_STATUS(self, value):
        encode_alphanumeric(self._data, 0, 2, value)
